### DISCUSSION TITLE
How to support code blocks without syntax highlighting?

### DISCUSSION THREAD

**Discussion by adaboese** [mdx-editor/editor#196] [Q&A] [answered]
If I just add:

```ts
codeBlockPlugin({
    defaultCodeBlockLanguage: 'js',
})
```

I get:

> No CodeBlockEditor registered for language=null meta=null

If I don't add plugin at all, I get:

> Error: no MdastImportVisitor found for code {"type":"code","lang":null,"meta":null,"value":"Before initiating

I just want simple code block without highlighting.

**Comment by skennedy100** [comment]
I've worked out a way to achieve this within the react framework, though it is not particularly pretty, it is based on the code in the [code blocks sections](https://mdxeditor.dev/editor/docs/code-blocks#:~:text=SandpackPreset%20interface.-,Build%20a%20custom%20code%20block%20editor,-You%20can%20implement):
```
const PlainTextCodeEditorDescriptor = {
  match: (language, meta) => true,
  priority: 0,
  Editor: props => {
    const textareaStyle = {
      fontFamily: "monospace",
      fontSize: "14px",
      padding: "10px",
      border: "1px solid #ccc",
      borderRadius: "5px",
      width: "100%",
      minHeight: "14px",
      height: "auto",
      backgroundColor: "#f4f4f4",
      resize: "vertical",
      outline: "none",
    }

    const textareaRef = useRef(null)
    useEffect(() => {
      // Adjust the height of the textarea based on its content
      const adjustHeight = () => {
        if (textareaRef.current) {
          textareaRef.current.style.height = "auto"
          textareaRef.current.style.height = `${
            textareaRef.current.scrollHeight + 14
          }px`
        }
      }

      adjustHeight()

      if (textareaRef.current) {
        textareaRef.current.addEventListener("input", adjustHeight)
      }

      return () => {
        if (textareaRef.current) {
          textareaRef.current.removeEventListener("input", adjustHeight)
        }
      }
    }, [])

    const cb = useCodeBlockEditorContext()

    return (
      <div onKeyDown={e => e.nativeEvent.stopImmediatePropagation()}>
        <textarea
          ref={textareaRef}
          style={textareaStyle}
          rows={3}
          defaultValue={props.code}
          onChange={e => cb.setCode(e.target.value)}
        />
      </div>
    )
  },
}
```

You can then use this descriptor directly with the codeBlockPlugin:
```
            codeBlockPlugin({
              codeBlockEditorDescriptors: [PlainTextCodeEditorDescriptor],
            }),
```

**Comment by kurtmkurtm** [comment, marked as answer]
If you are using the codeMirrorPlugin, you could try adding an empty string to the languages.
This fixed the same error for me

``` tsx
codeBlockPlugin(),
codeMirrorPlugin({
  codeBlockLanguages: {
    js: "JavaScript",
    txt: "text",
    "": "text",
  },
}),
```

**Comment by gopal-virtual** [comment]
Getting an error after adding an empty key. `A <Select.Item /> must have a value prop that is not an empty string.`

**Reply by petyosi** [reply]
@gopal-virtual ensure that you're on the latest version. I've explicitly fixed that problem here: 
https://github.com/mdx-editor/editor/commit/2d1d254fb8c055595c2db34d7af367688f63be98

**Reply by gopal-virtual** [reply]
While this works with the latest update, ideally it should fallback to `defaultCodeBlockLanguage` when no language are provided post backticks.
